library/ssl_tls13_client.c

```diff
@@ -1380,15 +1431,322 @@ static int ssl_tls13_finalize_server_hello( mbedtls_ssl_context *ssl )
     return( ret );
 }
 
+static int ssl_hrr_parse( mbedtls_ssl_context *ssl,
```

Comment:
It seems to me that but for the processing of some extensions (mainly MBEDTLS_TLS_EXT_KEY_SHARE) the code is very similar to that of `ssl_tls13_parse_server_hello()` thus probably that it would be better to just use `ssl_tls13_parse_server_hello()` to parse an HRR with some well placed `if( hrr )` or `if( ! hrr )`. Looking at it I am not even sure we need the "coordination" stage. We could just parse the server hello and check inside the parsing function if it is an HRR or not.